demo/custom.js

```diff
@@ -0,0 +1,265 @@
+/**
+ * Created by Samuel Gratzl on 25.08.2015.
+ */
+
+window.onload = function () {
+
+  var negative = 'true';
+
+  function getrandom(datapoints, min, max) {
+    var temparr = [];
+    for (var i = 0; i < datapoints; i++) {
+
+      if (negative == 'true') {
+        temparr.push(Math.floor(Math.random() * (max * 2 + 1)) - (max));
+      }
+      else {
+        temparr.push(Math.floor(Math.random() * (max - min) + min));
+
+      }
+    }
+
+    return (temparr);
+  }
+
+  var datapoints = 50, min = 0, max = 100;
+
+  var rand0 = getrandom(datapoints, min, max);
+  var rand1 = getrandom(datapoints, min, max);
+  var rand2 = getrandom(datapoints, min, max);
+  var rand3 = getrandom(datapoints, min, max);
+  var rand4 = getrandom(datapoints, min, max);
+
+
+  function catdata(datapoints) {
+    var temparr = [];
+    for (var i = 0; i < datapoints; i++) {
+
+      //temparr.push('c'+i,i);
+      temparr.push(Math.floor(Math.random() * 2));
+
+    }
+
+    return (temparr);
+  }
+
+
+  var data;
+  d3.json('data.json', function (data) {
+
+
+    var newdata = [];
+
+
+    data.reduce(function (a, b) {
```

Comment:
why did you use the 'reduce' function? you are not using the reduced value? are you looking for `map` or `forEach`